sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Common/src/Shared/Queues/QueueMetricsProvider.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Storage.Common.Listeners
+{
+    /// <summary>
+    /// Provides QueueTriggerMetrics from a specific queue entity.
+    /// </summary>
+    internal class QueueMetricsProvider
+    {
+        private readonly QueueClient _queue;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Instantiates a QueueMetricsProvider.
+        /// </summary>
+        /// <param name="queue">The QueueClient to use for metrics polling.</param>
+        /// <param name="loggerFactory">Used to create an ILogger instance.</param>
+        public QueueMetricsProvider(QueueClient queue, ILoggerFactory loggerFactory)
+        {
+            _queue = queue;
+            _logger = loggerFactory.CreateLogger<QueueMetricsProvider>();
+        }
+
+        // Caller of this function is responsible for wrapping this in try/catch
+        public async Task<int> GetQueueLength()
+        {
+            QueueProperties queueProperties = await _queue.GetPropertiesAsync().ConfigureAwait(false);
```

Comment:
do we need to add the same exception handling as in [GetMetricsAsync](https://github.com/Azure/azure-sdk-for-net/pull/32038/files#diff-32873ded99d2fb35e762e6579980552f4a7c235411f07940b565fc6e7633b425R70) to scale in?

I think the behavior should be consistent for incremental and target scale. I am going to change this for ServiceBus PR also.
this will a

Reply:
Agreed that exception handling behavior should be consistent for incremental and target scale. I added exception handling in `QueueTargetScaler.GetScaleResultAsync`, if `RequestFailedException` is thrown, then return 0 queue length to trigger scale in.

Reply:
I think the exception handling should happen in QueueMetricsProvider and use private helper method to check exception type and log. You can return 0 from GetQueueLength and it will translates "to scalein"